src/nyx/features/gabor.cpp

```diff
@@ -18,6 +21,17 @@ std::vector<std::pair<double, double>> GaborFeature::f0_theta_pairs
     {M_PI_4*3.0, 64.0} 
 };
 
+#ifdef USE_GPU
+    int GaborFeature::n_bank_filters = -1;
+    std::vector<std::vector<double>> GaborFeature::filterbank;
+    std::vector<double> GaborFeature::ho_filterbank;
+    double* GaborFeature::dev_filterbank = nullptr;

```

Comment:
The raw pointer here is used intentionally as its memory is allocated and deallocated via CUDA. (I let it be freed automatically on the process exit and because the work on Gabor GPU speedup is not over, accelerating the scoring part is the next effort.) Based on my experience, GPU code needs to be interfaced with as low-level objects as possible, meaning, no smart pointers, reliance on scope, virtual methods, etc. - check NVIDIA's own sample code as a good example and object-oriented code of OpenMP as a bad example.